src/hooks/logging.rs

```diff
@@ -0,0 +1,40 @@
+use crate::{EvaluationContext, EvaluationDetails, EvaluationError, Value};
+
+use super::{Hook, HookContext, HookHints};
+
+/// A hook that logs the evaluation lifecycle of a flag.
+pub struct LoggingHook;
+
+#[async_trait::async_trait]
+impl Hook for LoggingHook {
+    async fn before<'a>(
+        &self,
+        context: &HookContext<'a>,
+        _: Option<&'a HookHints>,
+    ) -> Result<Option<EvaluationContext>, EvaluationError> {
+        log::debug!("Before hook for flag {}", context.flag_key);
+
+        Ok(None)
+    }
+    async fn after<'a>(
+        &self,
+        context: &HookContext<'a>,
+        _: &EvaluationDetails<Value>,
+        _: Option<&'a HookHints>,
+    ) -> Result<(), EvaluationError> {
+        log::debug!("After hook for flag {}", context.flag_key);
+
+        Ok(())
+    }
+    async fn error<'a>(
+        &self,
+        context: &HookContext<'a>,
+        error: &EvaluationError,
+        _: Option<&'a HookHints>,
+    ) {
+        log::error!("Error hook for flag {}: {:?}", context.flag_key, error);
+    }
+    async fn finally<'a>(&self, context: &HookContext<'a>, _: Option<&'a HookHints>) {
```

Comment:
```suggestion
    async fn finally<'a>(&self, context: &HookContext<'a>, _: &EvaluationDetails<Value>, _: Option<&'a HookHints>) {
```

Hey @MaximFischuk, the `finally` stage is hooks changed slightly in a recent [spec update](https://github.com/open-feature/spec/pull/280). Thanks!

Reply:
Thank you for the information! Seems like I've missed it, I'll update it soon :)